src/iop/colorout.c

```diff
@@ -553,23 +575,39 @@ void process(struct dt_iop_module_t *self, dt_dev_pixelpipe_iop_t *piece, const
                                          ivoid, ovoid, roi_in, roi_out))
     return;
   const dt_iop_colorout_data_t *const d = (dt_iop_colorout_data_t *)piece->data;
-  const size_t width = roi_out->width;
-  const size_t height = roi_out->height;
-  const size_t npixels = width * height;
+  const size_t npixels = roi_in->width * roi_in->height;
```

Comment:
Since both fields are `int`, one of these should be cast to `size_t` to avoid truncation or overflow to negative on large images.

Reply:
Fully agreed :-)